### PR TITLE
Convert to dart Bigint 

### PR DESCRIPTION
**Pull request by wstrange**
A few changes here:
- BigInt conversion utilities moved into  ASN1Integer
- Some type fixes for dart 2:   intValue changed to an int getter (eliminated dynamic type)
- Split up unit test for the bigint tests

### CODE REVIEW

**Comment by wstrange** [comment]
@stevenroose The signature of ASN1Integer changes slightly  for stronger types. We will have to bump the sem version.

File: lib/asn1integer.dart [hunk 1]
```diff
@@ -3,86 +3,90 @@ part of asn1lib;
...
-  BigInteger get valueAsPositiveBigInteger {
-    if (!isEncoded) _encode();
-    return new BigInteger.fromBytes(1, valueBytes());
-  }
+  BigInt get valueAsBigInteger => _intValue;
 
   @override
   Uint8List _encode() {
-    var t = encodeIntValue(this.intValue);
+    var t = encodeBigIntValue(this._intValue);
```
Comment (stevenroose) [inline lib/asn1integer.dart]: For string mode, shouldn't this be `List<int>` instead of `var`?

File: lib/asn1integer.dart [hunk 2]
```diff
@@ -3,86 +3,90 @@ part of asn1lib;
...
    */
-  static Uint8List encodeIntValue(var intValue) {
-    var result;
-    if (intValue is BigInteger) {
-      result = new Uint8List.fromList(intValue.toByteArray());
-    } else {
-      var bn = new BigInteger(intValue);
-      result = new Uint8List.fromList(bn.toByteArray());
+
+  static BigInt bytes2BigInt(Uint8List bytes) {
```
Comment (stevenroose) [inline lib/asn1integer.dart]: Why `bytes2BigInt`, but `encodeBigIntValue`? I think a more consistent naming would be good. Just `encodeBigInt[Value]` and `decodeBigInt[Value]`.

File: lib/asn1integer.dart (hunk 1 above)
Comment (wstrange) [reply lib/asn1integer.dart]: strong mode? ok - makes sense

File: lib/asn1integer.dart (hunk 2 above)
Comment (wstrange) [reply lib/asn1integer.dart]: Yes - makes total sense.  I'm buried at work right now - but I'll make those changes

**Comment by wstrange** [comment]
Fixes #23

**Pull request merged**